ot3_state_manager/ot3_state_manager/messages.py

```diff
@@ -5,102 +5,156 @@
-from typing import Literal, Optional, Union
+from enum import Enum
+from typing import Any, Callable, Optional
 
 from opentrons.hardware_control.types import OT3Axis
 
 from ot3_state_manager.ot3_state import OT3State
-from ot3_state_manager.util import Direction, get_md5_hash
+from ot3_state_manager.util import Direction, MoveMessageHardware, SyncPinState
+
+MESSAGE_BYTE_LENGTH = 4
+MESSAGE_ID_BYTE_LENGTH = 1
+MESSAGE_CONTENT_BYTE_LENGTH = 3
+STRUCT_FORMAT_STRING = f"B{MESSAGE_CONTENT_BYTE_LENGTH}s"
+
+
+# mypy doesn't like that I am instantiating a dataclass based on an abstract class (ABC)
+# instead of a concrete class.
+@dataclass  # type: ignore[misc]
+class Message(ABC):
+    """Parent level class for all Message objects.
+
+    Is abstract base class that requires both a build and to_bytes method to be defined.
+    """
+
+    @staticmethod
+    def parse(message_bytes: bytes) -> Message:
+        """Parse sequence of bytes into a Message object."""
+        if not len(message_bytes) == MESSAGE_BYTE_LENGTH:
+            raise ValueError(
+                f"Message length must be {MESSAGE_BYTE_LENGTH} bytes. "
+                f"Your message was {len(message_bytes)} bytes."
+            )
+        message_id, message_content = struct.unpack(STRUCT_FORMAT_STRING, message_bytes)
+        # mypy thinks that I am instantiating a MoveMessageHardware object here.
+        # I am not, I am performing a lookup as defined in the enum docs.
+        # https://docs.python.org/3/library/enum.html#programmatic-access-to-enumeration-members-and-their-attributes
+        return MessageID(message_id).builder_func(message_content=message_content)  # type: ignore[call-arg]
+
+    @staticmethod
+    @abstractmethod
+    def build(message_content: bytes) -> Message:
+        """Parse message_content into a Message object."""
+        ...
+
+    @abstractmethod
+    def to_bytes(self) -> bytes:
+        """Convert Message object into a sequence of hexadecimal bytes."""
+        ...
 
 
 @dataclass
-class Message:
-    """Generic Message Type."""
+class MoveMessage(Message):
+    """Message for moving OT3 axis in specified direction."""
+
+    axis: OT3Axis
+    direction: Direction
 
-    @classmethod
-    def to_message(cls, string: str) -> Optional[Message]:
-        """Parse string into message."""
-        raise NotImplementedError
+    def __eq__(self, other: Any) -> bool:
+        """Confirm that the 2 objects are equal"""
+        return isinstance(other, MoveMessage) and (
+            self.axis == other.axis and self.direction == other.direction
+        )
+
+    def __hash__(self) -> int:
+        """Return hash for instance."""
+        return hash(str(self))
+
+    @staticmethod
+    def build(message_content: bytes) -> MoveMessage:
+        """Convert message_content into a MoveMessage object."""
+        hw_id, direction_val = struct.unpack(">HB", message_content)
+        # mypy thinks that I am instantiating a MoveMessageHardware object here.
```

Comment:
But those are the same thing. what's the error?

Reply:
The error is that the `axis` parameter is not defined.

Reply:
Removed `__new__` method and added helper functions to lookup by id. Making this a non-issue
26c23bb154ed560b05060954bc2d0f4c682ec017